pkg/cmd/roachtest/tests/sysbench.go

```diff
@@ -596,6 +623,101 @@ func exportSysbenchResults(
+			return roachtestutil.GetAllocsProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Collect 30 seconds of CPU profiles.
+	cpuProfiles, err := roachtestutil.GetProfileForAllCRDBNodes(c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetCPUProfile(ctx, c, t.L(), nodeID, time.Second*30 /* duration */)
+		})
+	require.NoError(t, err)
+
+	// Collect the second mutex profiles.
+	mutexProfiles2, err := roachtestutil.GetProfileForAllCRDBNodes(c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetMutexProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Collect the second allocs profiles.
+	allocsProfiles2, err := roachtestutil.GetProfileForAllCRDBNodes(
+		c,
+		t.L(),
+		func(nodeID int) (*profile.Profile, error) {
+			return roachtestutil.GetAllocsProfile(ctx, c, t.L(), nodeID)
+		})
+	require.NoError(t, err)
+
+	// Merge the CPU profiles into one profile that contains all the nodes.
+	cpuMerged, err := profile.Merge(cpuProfiles)
+	require.NoError(t, err)
+
+	// Calculate the difference between the two mutex profiles, and also between
+	// the two allocs profiles.
+	mutexDiffProfiles := make([]*profile.Profile, len(c.CRDBNodes()))
+	allocsDiffProfiles := make([]*profile.Profile, len(c.CRDBNodes()))
+	for i := range len(c.CRDBNodes()) {
+		mutexProfiles1[i].Scale(-1)
+		mutexDiffProfiles[i], err = profile.Merge([]*profile.Profile{mutexProfiles1[i],
+			mutexProfiles2[i]})
+		require.NoError(t, err)
+		mutexDiffProfiles[i].TimeNanos = mutexProfiles2[i].TimeNanos
+		mutexDiffProfiles[i].DurationNanos = mutexProfiles2[i].TimeNanos - mutexProfiles1[i].TimeNanos
+
+		allocsProfiles1[i].Scale(-1)
+		allocsDiffProfiles[i], err = profile.Merge([]*profile.Profile{allocsProfiles1[i],
+			allocsProfiles2[i]})
+		require.NoError(t, err)
+		allocsDiffProfiles[i].TimeNanos = allocsProfiles2[i].TimeNanos
+		allocsDiffProfiles[i].DurationNanos = allocsProfiles2[i].TimeNanos -
+			allocsProfiles1[i].TimeNanos
+	}
+
+	// Merge the mutex diff profiles into one profile that contains all the nodes.
+	mutexDiffMerged, err := profile.Merge(mutexDiffProfiles)
+	require.NoError(t, err)
+
+	// Do the same for the allocs diff profiles.
+	allocsDiffMerged, err := profile.Merge(allocsDiffProfiles)
+	require.NoError(t, err)
+
+	// exportProfile exports a profile to the artifacts' directory.
+	exportProfile := func(profile *profile.Profile, filename string) {
+		buf := bytes.Buffer{}
+		require.NoError(t, profile.Write(&buf))
+		err = os.WriteFile(filepath.Join(t.ArtifactsDir(), filename), buf.Bytes(), 0644)
+		require.NoError(t, err)
+	}
+
+	// Export all the profiles.
+	exportProfile(cpuMerged, "cpu-merged.prof")
+	exportProfile(mutexDiffMerged, "mutex-diff-merged.prof")
+	exportProfile(allocsDiffMerged, "allocs-diff-merged.prof")
+	for i := range len(c.CRDBNodes()) {
+		exportProfile(mutexDiffProfiles[i], fmt.Sprintf("mutex-diff-%d.prof", i+1))
```

Comment:
How about this schema throughout: `n1.mutex30s.pb.gz` (note that `pb.gz` is the canonical extension and I've had to rename files in the past so that tools would accept the file)